src/main/kotlin/lotto/model/Lotto.kt

```diff
@@ -1,10 +1,20 @@
 package lotto.model
 
-class Lotto private constructor(
+class Lotto(
     private val lotto: List<LottoNumber>
 ) {
 
-    fun findMatchedNumberCount(other: Lotto) = other.lotto.count { lotto.contains(it) }
+    init {
+        validateUnique(lotto)
+        validateNumberCount(lotto)
+    }
+
+    fun findRank(drawNumbers: DrawNumbers) =
+        WinningRank.of(findMatchedNumberCount(drawNumbers.winningNumbers), contains(drawNumbers.bonusNumber))
+
+    private fun findMatchedNumberCount(other: Lotto) = other.lotto.count { lotto.contains(it) }
```

Comment:
private 메서드를 만들때 확장함수를 활용해보시는 것도 좋을것 같습니다~!
상호작용 객체들이 어떤 객체들인지 바로 확인할 수 있을것 같습니다!

` WinningRank.of(lotto.findMatchedNumberCount1(drawNumbers.winningNumbers), contains(drawNumbers.bonusNumber))`

```suggestion
    private fun List<LottoNumber>.findMatchedNumberCount(other: Lotto) = other.lotto.count { this.contains(it) }
```